impl/src/main/java/org/pentaho/platform/genericfile/providers/repository/RepositoryFileProvider.java

```diff
@@ -441,4 +457,64 @@ private EnumSet<RepositoryFilePermission> getRepositoryPermissions( EnumSet<Gene
 
     return repositoryFilePermissions;
   }
+
+  @NonNull
+  @Override
+  public List<IGenericFile> getDeletedFiles() throws OperationFailedException {
+    return fileService.doGetDeletedFiles().stream()
+      .map( fileDto -> {
+        GenericFilePath parentPath = null;
+
+        try {
+          parentPath = GenericFilePath.parseRequired( fileDto.getPath() ).getParent();
+        } catch ( InvalidPathException e ) {
+          // noop
+        }
+
+        RepositoryObject repositoryObject =
+          convertFromNativeFileDto( fileDto, parentPath != null ? parentPath.toString() : null );
+
+        repositoryObject.setOwner( getOwnerFromNativeFileDto( fileDto ) );
```

Comment:
Shouldn't the owner always be filled in, even for non-deleted files?

Also, if the `getOwner()` of a deleted file does not match the original owner, then consider renaming `getOwnerFromNativeFileDto` to include it applies to deleted files only.

Reply:
It should 👍🏼 

But taking into account the [previous comment](https://github.com/pentaho/pentaho-generic-file-system/pull/8#discussion_r1952494892), should we start setting the owner for all files in this feature? we would need to expose a new flag `includeAcls` in order to `getOwner` return a value.

The deleted file owner isn't different, the problem is that the service doesn't expose that information, so we need to get it directly from the `unifiedRepository`

Reply:
What do you think about changing the method to:
```
private String getOwnerFromNativeFileDto( @NonNull RepositoryFileDto nativeFile ) {
    String owner = nativeFile.getOwner();
    if ( owner != null ) {
      return owner;
    }

    RepositoryFileAcl acl = unifiedRepository.getAcl( nativeFile.getId() );
    if ( acl != null ) {
      return acl.getOwner().getName();
    }

    return null;
  }
```